iroh-qrpc-types/src/p2p.rs

```diff
@@ -0,0 +1,482 @@
+use bytes::Bytes;
+use cid::Cid;
+use derive_more::{From, TryInto};
+use libp2p::{Multiaddr, PeerId};
+use quic_rpc::{
+    message::{Msg, RpcMsg, ServerStreaming},
+    Service,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+use crate::RpcResult;
+
+pub type P2pAddr = super::addr::Addr<P2pService>;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Key(pub Bytes);
+
+// rpc Version(google.protobuf.Empty) returns (VersionResponse) {}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct VersionRequest;
```

Comment:
any chance we can avoid having types for empty requests?

Reply:
Not easily. The purpose of the type is to distinguish between different requests. So you want two different enum cases.

Now, in principle you could write an enum that just has the request case without a value. But the current scheme works well with derive_more to automatically build you the two conversions that you need (From from the specific request to the request enum).

Also, I don't mind it that much.